evmrpc/filter.go

```diff
@@ -387,16 +389,16 @@ func (f *LogFetcher) GetLogsByFilters(ctx context.Context, crit filters.FilterCr
 		close(resultsChan) // Close the results channel after workers finish
 	}()
 
-	// Check err after all work is done
-	if len(errorsList) > 0 {
-		err = errors.Join(errorsList...)
-	}
-
 	// Aggregate results into the final slice
 	for result := range resultsChan {
 		res = append(res, result)
 	}
 
+	// Check err after all work is done
```

Comment:
This ordering adjustment make sense to me, but I wonder why it would lead to dead lock or stuck issue